root/tree/dataframe/RNTupleDSBenchmarks.cxx

```diff
@@ -0,0 +1,68 @@
+#include <ROOT/RNTupleDS.hxx>
+#include <ROOT/RDataFrame.hxx>
+
+#include <benchmark/benchmark.h>
+#include <string>
+#include <vector>
+
+#include <rootbench/RBConfig.h>
+
+namespace {
+
+constexpr double kKaonMassMeV = 493.677;
+
+double GetP2(double px, double py, double pz)
+{
+   return px*px + py*py + pz*pz;
+}
+
+double GetKE(double px, double py, double pz)
+{
+   double p2 = GetP2(px, py, pz);
+   return sqrt(p2 + kKaonMassMeV*kKaonMassMeV);
+}
+
+auto Dataframe(ROOT::RDataFrame &frame)
+{
+   auto fn_muon_cut = [](int is_muon) { return !is_muon; };
+   auto fn_k_cut = [](double prob_k) { return prob_k > 0.5; };
+   auto fn_pi_cut = [](double prob_pi) { return prob_pi < 0.5; };
+   auto fn_sum = [](double p1, double p2, double p3) { return p1 + p2 + p3; };
+   auto fn_mass = [](double B_E, double B_P2) { double r = sqrt(B_E*B_E - B_P2); return r; };
+
+   auto df_muon_cut = frame.Filter(fn_muon_cut, {"H1_isMuon"})
+                           .Filter(fn_muon_cut, {"H2_isMuon"})
+                           .Filter(fn_muon_cut, {"H3_isMuon"});
+   auto df_k_cut = df_muon_cut.Filter(fn_k_cut, {"H1_ProbK"})
+                              .Filter(fn_k_cut, {"H2_ProbK"})
+                              .Filter(fn_k_cut, {"H3_ProbK"});
+   auto df_pi_cut = df_k_cut.Filter(fn_pi_cut, {"H1_ProbPi"})
+                            .Filter(fn_pi_cut, {"H2_ProbPi"})
+                            .Filter(fn_pi_cut, {"H3_ProbPi"});
+   auto df_mass = df_pi_cut.Define("B_PX", fn_sum, {"H1_PX", "H2_PX", "H3_PX"})
+                           .Define("B_PY", fn_sum, {"H1_PY", "H2_PY", "H3_PY"})
+                           .Define("B_PZ", fn_sum, {"H1_PZ", "H2_PZ", "H3_PZ"})
+                           .Define("B_P2", GetP2, {"B_PX", "B_PY", "B_PZ"})
+                           .Define("K1_E", GetKE, {"H1_PX", "H1_PY", "H1_PZ"})
+                           .Define("K2_E", GetKE, {"H2_PX", "H2_PY", "H2_PZ"})
+                           .Define("K3_E", GetKE, {"H3_PX", "H3_PY", "H3_PZ"})
+                           .Define("B_E", fn_sum, {"K1_E", "K2_E", "K3_E"})
+                           .Define("B_m", fn_mass, {"B_E", "B_P2"});
+   auto hMass = df_mass.Histo1D<double>({"B_mass", "", 500, 5050, 5500}, "B_m");
+
+   return hMass;
+}
+} // namespace
+
+static void BM_RNTupleDS_LHCB(benchmark::State &state)
+{
+   auto df = ROOT::Experimental::MakeNTupleDataFrame("DecayTree", RB::GetDataDir() + "/B2HHH~none.ntuple");
+   auto h_ptr = Dataframe(df);
+   for (auto _ : state)
+      *h_ptr;
+   RB::Ensure(int(h_ptr->GetMean()) == 5262);
+   RB::Ensure(int(h_ptr->GetEntries()) == 23895);
+}
+BENCHMARK(BM_RNTupleDS_LHCB)->Unit(benchmark::kMicrosecond)->Iterations(1);
```

Comment:
I can't think of a way to run multiple event loops but keep the RDF setup out of the section measured by rootbench: after the first iteration, RDF just returns the computed results, does not run again.

I guess I could do this, but then we are also measuring the setup of the `Histo1D` action:
```suggestion
   for (auto _ : state)
      *df.Histo1D<double>({"B_mass", "", 500, 5050, 5500}, "B_m").GetValue();
   RB::Ensure(int(h_ptr->GetMean()) == 5262);
   RB::Ensure(int(h_ptr->GetEntries()) == 23895);
}
BENCHMARK(BM_RNTupleDS_LHCB)->Unit(benchmark::kMicrosecond)->Iterations(1);
```

Can I keep it at one iteration? runtime seems pretty stable anyway

Reply:
Can't we introduce an artificial column that changes a global state? After the first iteration when the flag is changed, we start measuring.  I think it's important to measure starting from the 2nd event in order to clip the initialization overhead.

Reply:
As per our discussion, since the RNTupleDS benchmark cannot easily ignore the runtime of the first event and the setup, I added instead a parameterization over the fraction of events processed and the benchmark is now run for 25%, 50% and 100% of the datapoints. From slope and offset of these measurements it's easy to figure out event processing rate as well as initialization offset.